### PR TITLE
Add IDL-index support (closes #159)

### LINKED ISSUE #159: Add interface summary to appendix
The old respec version allowed an interface summary:

```
<section class='appendix' id="webidl-ref">
  <h2>Interface summary</h2>
</section>
```

This is really helpful if a bigger interface gets split into different partial interface.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Gathers all the WebIDL into a single section. Keeps all the links.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@halindrome, wdyt?

**Comment by marcoscaceres** [comment]
Need to add tests before merging.

File: js/core/webidl-index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      if (idlIndexSec.title) {
+        header.innerHTML = idlIndexSec.title;
+        idlIndexSec.removeAttribute("title");
+      } else {
+        header.innerHTML = "IDL Index";
+      }
+      idlIndexSec.insertAdjacentElement("afterbegin", header);
+    }
+    if (!document.querySelector("pre.idl")) {
+      const text = "This specification doesn't declares any Web IDL.";
```
Comment (halindrome) [inline js/core/webidl-index.js]: doesn't declare

File: js/core/css/webidl-oldschool.css [hunk 2]
```diff
@@ -6,6 +6,14 @@ pre.idl {
     line-height:    120%;
 }
 
+.respec-idl-seperator {
+    padding: 0 0 0.4cm 0;
+}
+
+.respec-idl-seperator:last-child {
```
Comment (halindrome) [inline js/core/css/webidl-oldschool.css]: separator

File: js/core/webidl-index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        pre.classList.add("idl", "def");
+        Array
+          .from(document.querySelectorAll("pre.def.idl"))
+          .map(function(elem) {
+            return elem.cloneNode(true).firstElementChild;
+          })
+          .map(function(elem) {
+            const div = document.createElement("div");
+            div.appendChild(elem);
+            div.classList.add("respec-idl-seperator");
```
Comment (halindrome) [inline js/core/webidl-index.js]: separator

**Comment by halindrome** [comment]
Otherwise looks good and I think it is a good idea.

**Comment by marcoscaceres** [comment]
Wish I could spell proper and stuff.

**Pull request merged**